docs/usage/configuration.rst

```diff
@@ -0,0 +1,312 @@
+
+:code:`transition_debounce_window` (:ref:`duration<Duration>`)
+    If necessary, :command:`metricq_sink_nsca` tries to reduce the number of spurious :literal:`WARNING` or :literal:`CRITICAL` reports.
+    We call this process *"transition debouncing"*.
+    It does so by keeping a history of state transitions for each metric,
+    only sending an abnormal report if the majority of recent state transitions indicate an abnormal state.
+
+    Set this value to *enable* transition debouncing.
+    This configures how far into the past state transitions are kept in each history.
+
+    Default
+        Not set; transitions debouncing is disabled.
+
+    TODO
+        This should be called :literal:`transition_history_window`.
+        Bug me about it in an issue.
+
+:code:`plugins`
+    A dictionary of :ref:`plugin configurations<Plugin configuration>`.
+    Keys in this dictionary must match the regex :literal:`[a-z_]+`.
+
+
+NSCA host settings
+^^^^^^^^^^^^^^^^^^
+
+These settings tell the reporter where it should send its check results and how that host is configured.
+
+:code:`host` (`string`)
+    Address of the NSCA daemon to which check results are sent.
+    See :code:`-H` flag of :code:`send_nsca`.
+
+    Default:
+        :literal:`"localhost"`
+
+:code:`port` (`integer`)
+    Port of the NSCA daemon to which check results are sent.
+    See :code:`-p` flag of :code:`send_nsca`.
+
+    Default:
+        :literal:`5667`
+
+:code:`executable` (`string`)
+    Path to :code:`send_nsca` executable to use for sending check results.
+
+    Default:
+        :literal:`"/usr/sbin/send_nsca"`
+
+:code:`config_file` (`string`)
+    Path to :code:`send_nsca` configuration file.
+    See :code:`-c` flag of :code:`send_nsca`
+
+    Default:
+        :literal:`"/etc/nsca/send_nsca.cfg"`
+
+Plugin configuration
+^^^^^^^^^^^^^^^^^^^^
+
+:code:`file` (string)
+    File system path to plugin implementation (a `.py` file).
+
+    This key is **mandatory**.
+    Plugin configurations without it are rejected.
+
+:code:`config` (dictionary)
+    An arbitrary JSON object containing a plugin-specific configuration.
+
+Duration
+^^^^^^^^
+
+All durations in this configuration are *strings* in the form of :samp:`"{value} [{unit}]"`.
+The :samp:`{value}` is an integer or (decimal) floating point literal, the unit is one of
+
+* :literal:`s`/:literal:`seconds`
+* :literal:`ms`/:literal:`milliseconds`
+* :literal:`us`/:literal:`μs`/:literal:`microseconds`,
+* :literal:`ns`/:literal:`nanoseconds`
+* :literal:`d`/:literal:`days`
+* :literal:`h`/:literal:`hours`
+* :literal:`min`/:literal:`minutes`
+
+If the unit is not specified, the value is interpreted as a *number of seconds*.
+
+Examples
+    * :literal:`"5s"`, :literal:`"5"` (5 seconds)
+    * :literal:`"42 milliseconds"`
+    * :literal:`"1.5 days"`
+
+Complete Example
+----------------
+
+An example with all possible options set is given below:
```

Comment:
The example looks a bit borked. `plugins` should be part of the root object, don't they? It's also missing `resend_interval` and `reporting_host` in the root node.

Reply:
No, `plugins` are configured per-check.

But I'll add the other missing items to the example configuration.